File: 977-squares-of-a-sorted-array/977-squares-of-a-sorted-array.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        int n = size(nums);
        vector<int> ans(n);
        int i = 0, j = n - 1;
        for(int ptr = n - 1; ptr >= 0; --ptr) {
            if(abs(nums[j]) > abs(nums[i])) {
                ans[ptr] = nums[j] * nums[j];
                --j;
            } else {
                ans[ptr] = nums[i] * nums[i];
                ++i;
            }
        }
        return ans;
    }
};

File: 977-squares-of-a-sorted-array/977-squares-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        n = len(nums)
        ans = [0] * n
        i, j = 0, n - 1
        for ptr in range (n - 1, -1, -1):
            if abs(nums[j]) > abs(nums[i]):
                ans[ptr] = nums[j] * nums[j]
                j -= 1
            else:
                ans[ptr] = nums[i] * nums[i]
                i += 1
        return ans